std/transformer.glu

```diff
@@ -0,0 +1,11 @@
+//@NO-IMPLICIT-PRELUDE
+
+let { Monad } = import! std.prelude
+
+#[implicit]
+type Transformer t = {
```

Comment:
`Transformer` only seems to be used once in this code so maybe the abstraction isn't needed? Or can it be used to greater effect?

Reply:
Maybe `Transformer` could be factored out into its own PR, but I think if monad transformers are going to be included in the library, `Transformer` is an important abstraction to have. It and `wrap_monad` are equivalent to `MonadTrans` and `lift` in Haskell, where they are used liberally.

Reply:
Good point! Lets keep it 👍